iterators.tex

```diff
@@ -276,7 +276,7 @@
   template <class I>
   concept bool Readable@\newtxt{() \{}\oldtxt{ =}@
     @\newtxt{return}@ Semiregular<I>@\newtxt{()}@ &&
-      requires (I i) {
+      requires (@\newtxt{const }@I i) {
```

Comment:
...and now I'm waffling on this again. As far as Readable is concerned, the semantics of `*` are that it requires the value to be dereferencable and that it returns something convertible to `const ValueType<I>&`. If `*declval<T&>()`, `*declval<const T&>()`, `*declval<T&&>()`, and/or `declval<const T&&>()` return different types they can still be "semantically equivalent" if they are all non-modifying, return something convertible to `const ValueType<I>&`, and the values returned are not observably different to equality preserving expressions. In the case of `optional<U>`, the returned types would be `U&`, `const U&`, `U&&` and `const U&&` all bound to the same object. I think we can make `*i` non-modifying with the full complement of expression variants and not lose even Readables like `optional<U>` with deep constness.